nix/difftest/spike.nix

```diff
@@ -0,0 +1,47 @@
+{
+  stdenv,
+  dtc,
+  fetchFromGitHub,
+}:
+
+stdenv.mkDerivation {
+  version = "unstable-2024-07-03";
+  pname = "spike";
+
+  # env.cmakeConfig = ''
+  #   add_library(libspike STATIC IMPORTED GLOBAL)
+  #   set_target_properties(libspike PROPERTIES
+  #     IMPORTED_LOCATION "${placeholder "out"}/lib/libriscv.a")
+  #   target_include_directories(libspike AFTER INTERFACE
+  #     "${placeholder "out"}/include"
+  #     "${placeholder "out"}/include/riscv"
+  #     "${placeholder "out"}/include/fesvr"
+  #     "${placeholder "out"}/include/softfloat"
+  #   )
+  # '';
+  nativeBuildInputs = [ dtc ];
+  enableParallelBuilding = true;
+  separateDebugInfo = true;
+  src = fetchFromGitHub {
+    owner = "riscv";
+    repo = "riscv-isa-sim";
+    rev = "4a2da916671d49d9ab82f702f50995c19110c2a3";
```

Comment:
Maybe we should use spike at current HEAD `488e07d75df85d4bb363076dbb40b45288a2fad0`